include/parallel/parallel_implementation.h

```diff
@@ -3191,6 +3191,44 @@ void Communicator::scatter(const std::vector<T> & data,
 
 
 template <typename T>
+void Communicator::scatter(const std::vector<std::vector<T> > & data,
+                           std::vector<T> & recv,
+                           const unsigned int root_id,
+                           const bool identical_buffer_sizes) const
+{
+  if (this->size() == 1)
+    {
+      libmesh_assert (!this->rank());
+      libmesh_assert (!root_id);
+      return;
+    }
+
+  std::vector<T> stacked_data;
+  std::vector<int> counts;
+
+  if (root_id == this->rank())
+    {
+      libmesh_assert (data.size() == this->size());
+
+      if (!identical_buffer_sizes)
+        counts.resize(this->size());
+      for (unsigned int i=0; i < data.size(); ++i)
+        {
+          if (!identical_buffer_sizes)
+            counts[i] = data[i].size();
+          std::copy(data[i].begin(), data[i].end(), std::back_inserter(stacked_data));
+        }
+    }
+
+  if (identical_buffer_sizes)
+    this->scatter(stacked_data, recv, root_id);
```

Comment:
We should probably have a loop (wrapped in ifndef NDEBUG) which asserts that the user wasn't lying about the identical buffer sizes.